client/ts/display_difference_chart.ts

```diff
@@ -42,62 +44,136 @@ function clickHighlight(clickedElementList, nodes, sourceIndex, targetIndex, cli
+    imageElement.removeAttribute("height");
+    imageElement.removeAttribute("width");
+}
+
+
+function loadPNGGraph(sourceName:string, targetName:string, sourceIndex:number, targetIndex:number, clickedElement:HTMLElement){
+    let clickedElementList = [];
+    let imageType = '.png';
+    let staticImageFileName = sourceName + "_" + targetName + imageType;
+    let uploadFolderPath = '/static/upload_files/';
+
+    let staticImageURL = uploadFolderPath + staticImageFileName;
+    let imageElement:HTMLImageElement = document.getElementById("difference_chart") as HTMLImageElement;
+    let nodes = d3.selectAll("circle");
+
+    // let imageTitle = sourceName.toUpperCase() + ' VS ' + targetName.toUpperCase();
+    let imageTitle = sourceName.split("_").join(" ").toUpperCase() + ' VS ' + targetName.split("_").join(" ").toUpperCase();
+    Drawer.getInstance().open({mode:'plot', title: imageTitle});
 
+    $.ajax({
+        url: staticImageURL,
+        success: function(){
+            setImage(imageElement, staticImageURL);
+
+            // highlight currently clicked nodes, turn the previous clicked items into original color
+            clickHighlight(clickedElementList, nodes, sourceIndex, targetIndex, clickedElement);
+
+        },
+        error: function(){
+            // highlight currently clicked nodes, turn the previous clicked items into original color
+            clickHighlight(clickedElementList, nodes, sourceIndex, targetIndex, clickedElement);
+
+            // todo: in the future, using another way to handle missing pictures?
+            let errorMsg:string = "No Image For this Edge";
+            Drawer.getInstance().open({mode:'plot', title: errorMsg});
+
+            clearImage(imageElement);
         }
     });
+}
 
-}
\ No newline at end of file
+/**
+ *
+ * @param {string} type
+ * @param {string} sourceName
+ * @param {string} targetName
+ * @param {number} sourceIndex
+ * @param {number} targetIndex
+ * @param {HTMLElement} clickedElement
+ */
+function loadPlotylPlot(type:string, sourceName:string, targetName:string, sourceIndex:number, targetIndex:number, clickedElement:HTMLElement){
+    loadJson(sourceName, targetName,
+        function (data){
+            let imageTitle = sourceName.split("_").join(" ").toUpperCase() + ' VS ' + targetName.split("_").join(" ").toUpperCase();
+            Drawer.getInstance().open({mode:'plot', title: imageTitle});
+            createPlot(type, data, 'plot-parent');
+        },
+        function () {
+            Drawer.getInstance().open({mode:'plot', title: "Could not load data to plot."});
+        });
+
+    let clickedElementList = [];
+    let nodes = d3.selectAll("circle");
+    clickHighlight(clickedElementList, nodes, sourceIndex, targetIndex, clickedElement);
+}
+
+
+/**
+ * Tries to load the JSON by alternating node names.
+ * It is not clear which name should be first from the graph.
+ * @param {string} sourceName
+ * @param {string} targetName
+ * @param onComplete
```

Comment:
what is onComplete and onError? Are they customized call back function?

Reply:
huh i think i see them  now

Reply:
Added to documentation.